controllers/priorityexpander_controller.go

```diff
@@ -0,0 +1,240 @@
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the PriorityExpander object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *PriorityExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx, "pexp", req.NamespacedName, "reconciliationID", uuid.New().String())
+
+	log.V(1).Info("PriorityExpander reconciliation started")
+	defer log.V(1).Info("PriorityExpander reconciliation done")
+
+	var pexp corev1alpha1.PriorityExpander
+	if err := r.Get(ctx, req.NamespacedName, &pexp); err != nil {
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		err = client.IgnoreNotFound(err)
+		if err != nil {
+			log.Error(err, "Unable to fetch PriorityExpander")
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	// Fetch clusterautoscaler status configmap.
+	var cm kcore_v1.ConfigMap
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: r.Configuration.ClusterAutoscalerStatusNamespace,
+		Name:      r.Configuration.ClusterAutoscalerStatusName,
+	}, &cm); err != nil {
+		log.Error(
+			err,
+			"Unable to fetch ClusterAutoscaler status configmap",
+			"namespace", r.Configuration.ClusterAutoscalerStatusNamespace,
+			"name", r.Configuration.ClusterAutoscalerStatusName,
+		)
+
+		return ctrl.Result{}, err
+	}
+	// Get human readable status from configmap...
+	readableStatus, ok := cm.Data["status"]
+	if !ok {
+		err := fmt.Errorf("invalid configmap: no status")
+		log.Error(
+			err,
+			"unable to parse ClusterAutoscaler status",
+			"namespace", r.Configuration.ClusterAutoscalerStatusNamespace,
+			"name", r.Configuration.ClusterAutoscalerStatusName,
+		)
+	}
+
+	// ... and parse it.
+	status := clusterautoscaler.ParseReadableString(readableStatus)
+
+	var oroot = map[string]map[string]int32{}
+	for _, node := range status.NodeGroups {
+		oroot[node.Name] = make(map[string]int32)
+		oroot[node.Name]["CloudProviderTarget"] = node.Health.CloudProviderTarget
+		oroot[node.Name]["Ready"] = node.Health.Ready
+		oroot[node.Name]["Unready"] = node.Health.Unready
+		oroot[node.Name]["NotStarted"] = node.Health.NotStarted
+		oroot[node.Name]["LongNotStarted"] = node.Health.LongNotStarted
+		oroot[node.Name]["Registered"] = node.Health.Registered
+		oroot[node.Name]["LongUnregistered"] = node.Health.LongUnregistered
+		oroot[node.Name]["MinSize"] = node.Health.MinSize
+		oroot[node.Name]["MaxSize"] = node.Health.MaxSize
+	}
+
+	//Create new PriorityExpander template and parse it
+	t, err := template.New("template").Parse(pexp.Spec.Template)
+	if err != nil {
```

Comment:
If the template is not correct, the reconciliation will fail indefinitely.
It would be good to indicate that to the user (via the status of the PriorityExpander for example).

Reply:
as discussed, having a Failed state wouldn't change much.